cooperative_lanechange/include/cooperative_lanechange.h

```diff
@@ -251,9 +251,21 @@ namespace cooperative_lanechange
              * \param tf The transform between the world frame and map frame in which the trajectory plan points are calculated
              * \return The trajectory point message transformed to world frame
              */
-            cav_msgs::LocationECEF trajectory_point_to_ecef(const cav_msgs::TrajectoryPlanPoint& traj_point, const geometry_msgs::TransformStamped& tf) const;
+            cav_msgs::LocationECEF trajectory_point_to_ecef(const cav_msgs::TrajectoryPlanPoint& traj_point, const tf2::Transform& transform) const;
 
             void add_maneuver_to_response(cav_srvs::PlanTrajectoryRequest &req, cav_srvs::PlanTrajectoryResponse &resp, std::vector<cav_msgs::TrajectoryPlanPoint>& planned_trajectory_points);
+            
+                          /**
+             * \brief Given the curvature fit, computes the curvature at the given step along the curve
+             * 
+             * \param step_along_the_curve Value in double from 0.0 (curvature start) to 1.0 (curvature end) representing where to calculate the curvature
+             * 
+             * \param fit_curve curvature fit
+             * 
+             * \return Curvature (k = 1/r, 1/meter)
+             */ 
+            double compute_curvature_at(const cooperative_lanechange::smoothing::SplineI& fit_curve, double step_along_the_curve) const;
```

Comment:
I'll write up a user story to finish the basic autonomy refactor for this plugin.